### PR TITLE
Add `Rootstock`  chain Testnet

### PR DESCRIPTION
**Pull request by ahsan-javaiid**
### Description

This PR adds Rootstock testnet chain to registry. Contracts deployed and tested sending / receiving message using rootstock testnet chain. 

Further plans to add rootstock mainnet.

## About Rootstock
Website: https://rootstock.io/
Defillama: https://defillama.com/chain/Rootstock
Explorer: https://explorer.rootstock.io/
Developer portal: https://dev.rootstock.io/

### Backward compatibility

Are these changes backward compatible? Note that additions are backwards compatible.

Yes

### Testing

Tested sending / receiving messages to / from rootstock chain.

### CODE REVIEW

**Comment by ahsan-javaiid** [comment]
Hi @jmrossy

Created this PR to add Rootstock testnet. After that plan to add mainnet. Feel free to comment if anything further needs to be added or changed.

File: chains/rsktestnet/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  estimateBlockTime: 30
+  reorgPeriod: 4
+chainId: 31
+deployer:
+  name: Rootstock
+  url: https://rootstock.io
+displayName: Rootstock Testnet
+domainId: 31
+isTestnet: true
+name: rootstocktestnet
```
Comment (jmrossy) [inline chains/rsktestnet/metadata.yaml:18]: The folder name must match this string, so `rootstocktestnet`

File: chains/rsktestnet/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+nativeToken:
+  decimals: 18
+  name: tRBTC
+  symbol: tRBTC
+protocol: ethereum
+rpcUrls:
+  - http: https://public-node.testnet.rsk.co
+  - http: https://rpc.testnet.rootstock.io/AluZdy8x1eQCwbFLmeLm7SUbxMQ1vs-T
```
Comment (jmrossy) [inline chains/rsktestnet/metadata.yaml:26]: This looks like an API key. Are you sure you want this included?

**Review by jmrossy — COMMENTED** [review]
Thanks for the contribution! Just a few small things to fix

File: chains/rsktestnet/metadata.yaml (hunk 1 above)
Comment (ahsan-javaiid) [reply chains/rsktestnet/metadata.yaml]: Fixed! Thanks for pointing it out.

File: chains/rsktestnet/metadata.yaml (hunk 2 above)
Comment (ahsan-javaiid) [reply chains/rsktestnet/metadata.yaml]: Fixed! Thanks for pointing it out.

**Comment by jmrossy** [comment]
@ahsan-javaiid it looks like the RPC URL you have provided isn't working according to the registry's RPC health test:

Failure details in CI:
```
1) Chain RPC health
       rootstocktestnet RPC number 0 is healthy:
     Error: bad response (status=403, headers={"date":"Fri, 26 Jul 2024 00:31:02 GMT","content-type":"text/html; charset=utf-8","content-length":"280","connection":"close","server":"Varnish","x-varnish":"116701641","retry-after":"5"}, body="<!DOCTYPE html>\n<html>\n  <head>\n    <title>403 Method Not Allowed</title>\n  </head>\n  <body>\n    <h1>Error 403 Method Not Allowed</h1>\n    <p>Method Not Allowed</p>\n    <h3>Guru Meditation:</h3>\n    <p>XID: 116701641</p>\n    <hr>\n    <p>Varnish cache server</p>\n  </body>\n</html>\n", requestBody="{\"method\":\"eth_getLogs\",\"params\":[{\"fromBlock\":\"0x51e1a3\",\"toBlock\":\"0x51e206\",\"address\":\"0xcfa3e807def506db480328cb975fc9108eb59e52\",\"topics\":[\"0x788dbc1b7152732178210e7f4d9d010ef016f9eafbe66786bd7169f56e0c353a\"]}],\"id\":45,\"jsonrpc\":\"2.0\"}", requestMethod="POST", url="https://public-node.testnet.rsk.co/", code=SERVER_ERROR, version=web/5.7.1)
      at Logger.Logger.makeError (node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
      at Logger.Logger.throwError (node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
      at <anonymous> (node_modules/@ethersproject/web/src.ts/index.ts:308:24)
      at step (node_modules/@ethersproject/web/src.ts/index.ts:1:14)
      at Object.next (node_modules/@ethersproject/web/src.ts/index.ts:1:14)
      at fulfilled (node_modules/@ethersproject/web/src.ts/index.ts:1:14)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by ahsan-javaiid** [comment]
> @ahsan-javaiid it looks like the RPC URL you have provided isn't working according to the registry's RPC health test:
> 
> Failure details in CI:
> 
> ```
> 1) Chain RPC health
>        rootstocktestnet RPC number 0 is not healthy:
>      Error: bad response (status=403, headers={"date":"Fri, 26 Jul 2024 00:31:02 GMT","content-type":"text/html; charset=utf-8","content-length":"280","connection":"close","server":"Varnish","x-varnish":"116701641","retry-after":"5"}, body="<!DOCTYPE html>\n<html>\n  <head>\n    <title>403 Method Not Allowed</title>\n  </head>\n  <body>\n    <h1>Error 403 Method Not Allowed</h1>\n    <p>Method Not Allowed</p>\n    <h3>Guru Meditation:</h3>\n    <p>XID: 116701641</p>\n    <hr>\n    <p>Varnish cache server</p>\n  </body>\n</html>\n", requestBody="{\"method\":\"eth_getLogs\",\"params\":[{\"fromBlock\":\"0x51e1a3\",\"toBlock\":\"0x51e206\",\"address\":\"0xcfa3e807def506db480328cb975fc9108eb59e52\",\"topics\":[\"0x788dbc1b7152732178210e7f4d9d010ef016f9eafbe66786bd7169f56e0c353a\"]}],\"id\":45,\"jsonrpc\":\"2.0\"}", requestMethod="POST", url="https://public-node.testnet.rsk.co/", code=SERVER_ERROR, version=web/5.7.1)
>       at Logger.Logger.makeError (node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
>       at Logger.Logger.throwError (node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
>       at <anonymous> (node_modules/@ethersproject/web/src.ts/index.ts:308:24)
>       at step (node_modules/@ethersproject/web/src.ts/index.ts:1:14)
>       at Object.next (node_modules/@ethersproject/web/src.ts/index.ts:1:14)
>       at fulfilled (node_modules/@ethersproject/web/src.ts/index.ts:1:14)
>       at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> ```

@jmrossy 

I looked into the issue. `eth_getLogs` rpc method is mostly not supported on public nodes due to security reasons. Just asking is it a hard requirement for hyperlane / compulsory for the adding chain to support this on public nodes.?

**Comment by jmrossy** [comment]
@ahsan-javaiid Yes currently some `eth_getLogs` support is required because our SDK requires that method in a few places. Many public RPCs have limits on block range and that's fine. If that's the case, you can configure it the `pagination` and `concurrency` fields on the RPC config in your chain metadata: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/sdk/src/metadata/chainMetadataTypes.ts#L41

**Comment by ahsan-javaiid** [comment]
@jmrossy 

After internal discussion, we decided to use rpc api with "api key".  RPC api has support for eth_getLogs so we want to move forward with that. We are aware of the fact that api_key is publicly visible here. 

Would be great if you can retrigger the workflows to test the PR. Feel free to ping here if you have different thought's or anything needs to be changed.

File: .changeset/red-bottles-remain.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.changeset/red-bottles-remain.md
+
```
Comment (jmrossy) [inline .changeset/red-bottles-remain.md:2]: I don't think a changeset is meant to include these two lines at the top here

File: chains/rootstocktestnet/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+domainId: 31
+isTestnet: true
+name: rootstocktestnet
+nativeToken:
+  decimals: 18
+  name: tRBTC
+  symbol: tRBTC
+protocol: ethereum
+rpcUrls:
+  - http: https://rpc.testnet.rootstock.io/YZIZHxGCQAryubOWH7LjdtjNkSDb5J-T
```
Comment (jmrossy) [inline chains/rootstocktestnet/metadata.yaml:25]: This looks like an API key. Are you sure you meant to include it here in this public repo?

**Review by jmrossy — COMMENTED** [review]
Thanks, just a few small things to fix

File: .changeset/red-bottles-remain.md (hunk 3 above)
Comment (ahsan-javaiid) [reply .changeset/red-bottles-remain.md]: Fixed!

File: chains/rootstocktestnet/metadata.yaml (hunk 4 above)
Comment (ahsan-javaiid) [reply chains/rootstocktestnet/metadata.yaml:25]: @jmrossy Yes. We are sure to include it in public repo. We discussed and well thought to use api key here. 
Adding @alepc253 to this discussion.

Comment (ahsan-javaiid) [reply chains/rootstocktestnet/metadata.yaml:25]: @alepc253 👆

File: .changeset/red-bottles-remain.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@hyperlane-xyz/registry": patch
```
Comment (jmrossy) [inline .changeset/red-bottles-remain.md:2]:
```suggestion
"@hyperlane-xyz/registry": minor
```

**Review by jmrossy — APPROVED** [review]

File: chains/rootstocktestnet/metadata.yaml (hunk 4 above)
Comment (jmrossy) [reply chains/rootstocktestnet/metadata.yaml:25]: A comment would be a good idea! Thankfully this is YAML, not JSON, so you can add any comments you like above this line

Comment (ahsan-javaiid) [reply chains/rootstocktestnet/metadata.yaml:28]: Added comments. Thanks for this suggestion. 👍

File: .changeset/red-bottles-remain.md (hunk 5 above)
Comment (ahsan-javaiid) [reply .changeset/red-bottles-remain.md]: ✅

**Comment by jmrossy** [comment]
@ahsan-javaiid seeing a few CI failures. Lmk if you'd like help with these

**Comment by ahsan-javaiid** [comment]
> @ahsan-javaiid seeing a few CI failures. Lmk if you'd like help with these

Updated the pull request. Please retrigger the workflows. ✅

**Comment by ahsan-javaiid** [comment]
Thank you @jmrossy for helping and considering the PR. I will send next one with mainnet. Thank you 🤩

**Pull request merged**